rclcpp/test/rclcpp/test_qos_event.cpp

```diff
@@ -109,6 +119,11 @@ TEST_F(TestQosEvent, test_publisher_constructor)
  */
 TEST_F(TestQosEvent, test_subscription_constructor)
 {
+  std::string rmw_implementation_str = std::string(rmw_get_implementation_identifier());
```

Comment:
While `rmw_zenoh` does not support `Deadline/LivelinessChanged` events, it does uspport `IncompatibleQoS`. So instead of skipping this entire test, can we modify the code so that the `options.event_callbacks.deadline_callback` and `options.event_callbacks.liveliness_callback` are not assigned if `rmw_implementation_str == "rmw_zenoh_cpp"`?